nginx/Dockerfile.tmpl

```diff
@@ -32,21 +32,30 @@ RUN apk update && \
       nginx-mod-http-lua \
       nginx-mod-http-echo \
       nginx-mod-http-cache-purge \
-      nginx-mod-http-fancyindex && \
+      nginx-mod-http-fancyindex \
+      nginx-mod-http-headers-more && \
     rm -rf /var/cache/apk/* && \
     mkdir -p  /etc/nginx/env /run/nginx /srv/www/html /var/log/nginx /var/cache/nginx && \
     sed -i 's/access_log.*/access_log off;/' /etc/nginx/nginx.conf && \
-    mv /tmp/default.conf /etc/nginx/conf.d/default.conf && \
+    mv /tmp/default.conf /etc/nginx/http.d/default.conf && \
     mv /tmp/env.conf /etc/nginx/env/env.conf && \
     mv /tmp/index.html /srv/www/html/index.html && \
+    sed -i "s/^http {/http {\n\tmore_set_headers 'X-Content-Type-Options: nosniff';\n/" /etc/nginx/nginx.conf && \
+    sed -i "s/^http {/http {\n\tmore_set_headers 'X-XSS-Protection: 1; mode=block';/" /etc/nginx/nginx.conf && \
+    sed -i "s/^http {/http {\n\tmore_set_headers 'X-Frame-Options: SAMEORIGIN';/" /etc/nginx/nginx.conf && \
+    sed -i "s/^http {/http {\n\tmore_set_headers \"Content-Security-Policy: default-src 'self'\";/" /etc/nginx/nginx.conf && \
```

Comment:
The default value for many other sources is `default-src` — see here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src

I think if we set a CSP without declaring `unsafe-inline` for JS, it will immediately invalidate most of our NewRelic implementations, and GTM as well. In Drupal the NewRelic code gets inlined at the very top of the document. GTM is typically inlined.

In both cases there are dynamic values that we pass in, meaning we can't add a one-size-fits-all hash that would allow the inlined code to slip through.

Reply:
@teodorescuserban in general, you should be very wary of doing something so broadly like `default-src` because as written, this will cause severe disruption to every property we manage. NewRelic, GTM, sharing buttons, all video embeds, all Google fonts, requests to CDNs, XHR to external services like MailChimp.... the list goes on.

Reply:
@rupl this should be a baseline from which all properties can move left or right. If we do see the majority of the sites in need of a specific change, then sure, I am happy to put _that_ instead as default and only change the remaining stacks. 
Thank you for your feedback and looking forward to see some proposals specific for properties (hid, dsr, etc). As I mentioned in the corresponding tickets, these default values _can_ be overwritten, but let;s test them first and come up with property-specific changes.

Reply:
Piping in on the comments from @rupl  about NewRelic and Google Tag Manager. Both of these should be considered "required" for sites in our shared infra. We may as well add them to the "allowed" list in the base image. Google Fonts is another "required" for all sites using the Common Design (which should be all of them).

Reply:
Just for the record, no Drupals use the nginx image, they all use the unocha/php-k8s image, which is a unocha/php image with nginx added on.

All Drupals can manage all these headers themselves via the seckit module, with the configuration stashed in config/yaml in the repository or in Ansible.